packages/taquito-local-forging/src/taquito-local-forging.ts

```diff
@@ -44,6 +47,17 @@ export class LocalForger implements Forger {
   private codec = getCodec(CODEC.MANAGER, this.protocolHash);
 
   forge(params: ForgeParams): Promise<string> {
+    if (validateBlock(params.branch) !== ValidationResult.VALID) {
+      throw new InvalidBlockHashError(`The block hash ${params.branch} is invalid`);
+    }
+
+    params.contents.forEach((e) => {
+      const diff = validateMissingProperty(e);
+      if (diff.length > 0 && diff[0] !== 'delegate' && diff[0] !== 'parameters') {
```

Comment:
I understand that `parameters` are optional in `transaction` operation and `delegate` is optional in `origination` and `delegation` operations. For code clarity, should we check for those missing parameters specifically when opKind is one of the above?

Also, I think that because script comes after delegate in the OriginationSchema, there won't be an exception thrown if script and delegate are missing (even if the script is required because delegate will be at position 1 of the diff array). We can add a test case for that scenario.

Reply:
I agree with both cases, and great catch on the delegate/script being missing, I did not think of that. Will make amendments